web/modules/custom/server_general/src/NodeViewBuilderAbstract.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+namespace Drupal\server_general;
+
+use Drupal\Core\Block\BlockManagerInterface;
+use Drupal\Core\Entity\EntityTypeManager;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\node\NodeInterface;
+
+/**
+ * Class NodeViewBuilderPerBundleAbstract.
+ */
+class NodeViewBuilderAbstract {
+
+  use ComponentWrapTrait;
+
+  /**
+   * The image style to use on Hero images.
+   */
+  const IMAGE_STYLE_HERO = 'hero';
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $currentUser;
+
+  /**
+   * The block manager.
+   *
+   * @var \Drupal\Core\Block\BlockManagerInterface
+   */
+  protected $blockManager;
+
+  /**
+   * Abstract constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManager $entity_type_manager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Session\AccountInterface $current_user
+   *   The current user.
+   * @param \Drupal\Core\Block\BlockManagerInterface $block_manager
+   *   The block manager.
+   */
+  public function __construct(EntityTypeManager $entity_type_manager, AccountInterface $current_user, BlockManagerInterface $block_manager) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->currentUser = $current_user;
+    $this->blockManager = $block_manager;
+  }
+
+  /**
+   * Default build in "Teaser" view mode.
+   *
+   * Show nodes as "cards".
+   *
+   * @param array $build
+   *   The existing build.
+   * @param \Drupal\node\NodeInterface $entity
+   *   The entity.
+   *
+   * @return mixed[]
+   *   An array of elements for output on the page.
+   */
+  public function buildTeaser(array $build, NodeInterface $entity) {
+    $build += $this->getElementBase($entity);
+    $build['#theme'] = 'server_theme_card__simple';
+
+    return $build;
+  }
+
+  /**
+   * Get common elements for the view modes.
+   *
+   * @param \Drupal\node\NodeInterface $entity
+   *   The entity.
+   *
+   * @return array[]
```

Comment:
```suggestion
   * @return array
```

or 

```suggestion
   * @return mixed[]
```

(id is an int I think, others are string)